.github/workflows/codespaces.yml

```diff
@@ -0,0 +1,87 @@
+name: GitHub Codespaces image build
+
+# Run only on master branch. We could also build on branch but this seems like
+# an optimization that can be done as and when desired. The main use case we're
+# handling is creating and working off a branch from master, so it doesn't seem
+# like an immediate requirement to have branches as well.
+#
+# NOTE: the job is setup to also push branch images as well, and using branch
+# and master as caching, so if we want to add the optimisation for branches we
+# can just remove the master branch restriction.
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        types:
+            - opened
+            - labeled
+            - synchronize
+
+jobs:
+    build:
+        name: Build Codespaces image
+        runs-on: ubuntu-latest
+
+        # Build on master and PRs with the label 'codespaces-build' only
+        if: ${{ github.ref == 'refs/heads/master' || contains(github.event.pull_request.labels.*.name, 'codespaces-build')  }}
+
+        steps:
+            - uses: actions/checkout@v1
+              with:
+                  fetch-depth: 1
+
+            # As ghcr.io complains if the image has upper case letters, we use
+            # this action to ensure we get a lower case version. See
+            # https://github.com/docker/build-push-action/issues/237#issuecomment-848673650
+            # for more details
+            - name: Docker image metadata
+              id: meta
+              uses: docker/metadata-action@v3
+              with:
+                  images: ghcr.io/${{ github.repository }}/codespaces
```

Comment:
ditto

Reply:
Yup same, they're only relevant within github so thought I'd just check into ghcr, without polluting the public facing dockerhub page.